src/hooks/useDebouncedFunction.ts

```diff
@@ -5,10 +5,13 @@ type Func = (...args: any[]) => any;
 type DebouncedFunction<F extends Func> = (...args: Parameters<F>) => Promise<undefined | ReturnType<F>>;
 
 export function useDebouncedFunction<F extends Func>(
-  func: F,
+  func: F | undefined,
```

Comment:
instead of letting this be undefined, could we just check in the Map component if onDrag is defined before using this hook?

Reply:
React Hooks can't be called conditionally

Reply:
ah right